src/bot/commands.rs

```diff
@@ -65,16 +103,78 @@ pub trait Command {
         api: &Api,
     ) -> String;
 
-    fn execute(&self, db_pool: Pool<ConnectionManager<PgConnection>>, api: Api, message: Message) {
+    fn execute(
+        &self,
+        db_pool: Pool<ConnectionManager<PgConnection>>,
+        api: Api,
+        mut message: Message,
+    ) {
         info!(
             "{:?} wrote: {}",
             message.chat.id,
             message.text.as_ref().unwrap()
         );
 
-        let text = self.response(db_pool, &message, &api);
+        let text: &str = message.text.as_ref().unwrap();
+        let command = Commands::from_str(text).unwrap();
+        let data = match self.fetch_db_connection(db_pool.clone()) {
+            Ok(mut connection) => self.list_subscriptions(&mut *connection, message.clone()),
+            Err(_error_message) => "error fetching data".to_string(),
+        };
+
+        let feed_id = match self.fetch_db_connection(db_pool.clone()) {
+            Ok(mut connection) => self.list_feed_id(&mut *connection, &message),
+            Err(_error_message) => "error fetching data".to_string(),
+        };
+
+        let feeds = data.split('\n');
+        let feeds_ids = feed_id.split(',').clone();
+        let text = self.response(db_pool.clone(), &message, &api);
+        let delete_message_params = DeleteMessageParams::builder()
+            .chat_id(message.chat.id)
+            .message_id(message.message_id)
+            .build();
 
-        self.reply_to_message(api, message, text)
+        match command {
```

Comment:
this is very hacky way to intercept the command. let's modify `list_subscriptions` instead